sdks/java/harness/src/main/java/org/apache/beam/fn/harness/WindowMergingFnRunner.java

```diff
@@ -151,10 +151,12 @@ public void merge(Collection<W> toBeMerged, W mergeResult) throws Exception {
         KV<T, Iterable<W>> windowsToMerge) throws Exception {
       currentWindows = Sets.newHashSet(windowsToMerge.getValue());
       windowFn.mergeWindows((MergeContext) mergeContext);
```

Comment:
We can do `mergedWindows.clear();` before we invoke `windowFn.mergeWindows` instead of making a copy, right?

Reply:
You are right. Theoretically we don't have to store the state and wait for next merge window invocation to clean up, I don't have strong opinion on this, I guess only storing last merge results won't take too much memory space.

Reply:
makes sense.